### PR TITLE
Collect simulator diagnostics

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Context

In case of simulator failures, this option enables collecting data to be forwarded to Apple.

Resolves: https://bitrise.atlassian.net/browse/STEP-707

### Changes

- Start simulator manually using _simctl_ tool, this is similar to the headless mode that Xcode defaults to. More logs can be collected from a running simulator, this way it is kept running after _xcodebuild test_ finishes.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ workflows:
     - BITRISE_PROJECT_PATH: ios-simple-objc/ios-simple-objc.xcodeproj
     - BITRISE_SCHEME: ios-simple-objc
     - SIMULATOR_DEVICE: iPhone 8 Plus
-    - SIMULATOR_OS_VERSION: latest
+    - SIMULATOR_OS_VERSION: "latest"
```
Comment (godrei) [inline bitrise.yml:53]: what is the reason for this change?

File: main.go [hunk 2]
```diff
@@ -558,6 +567,11 @@ func main() {
 		log.Warnf("Export UITest Artifacts (export_uitest_artifacts) turned on, but Xcode version >= 11. The test result bundle structure changed in Xcode 11 it does not contain TestSummaries.plist and Attachments directly, nothing to export.")
 	}
 
+	if simulatorDebug != never && xcodeMajorVersion < 10 {
+		log.Warnf("Collecting Simulator diagnostics is available with Xcode version 10 or newer")
```
Comment (godrei) [inline main.go:571]: Collecting Simulator diagnostics is not available below Xcode version 10, current Xcode version: %s

File: simulator.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/errorutil"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/ziputil"
+)
+
+func simulatorBoot(id string) error {
```
Comment (godrei) [inline simulator.go:18]: What is the difference between this simulator boot and `simulator.BootSimulator`?
Should not the content of this file go to that `simulator` package?

File: main.go [hunk 4]
```diff
@@ -646,9 +660,23 @@ func main() {
 		buildTestParams.CleanBuild = configs.IsCleanBuild
 	}
 
+	if simulatorDebug != never {
+		log.Infof("Enabling Simulator verbose log for better diagnostics")
+		// Boot the simulator now, so verbose logging can be enabled and it is kept booted after running tests,
+		// this helps to collect more detailed debug info
+		if err := simulatorBoot(sim.ID); err != nil {
```
Comment (godrei) [inline main.go:667]: is it possible we try to boot the simulator some cases twice? (simulatorDebug != never && sim.Status == simulatorShutdownState && !configs.HeadlessMode, line 682)

**Review by godrei — CHANGES_REQUESTED** [review]

File: bitrise.yml [hunk 5]
```diff
@@ -239,6 +240,7 @@ workflows:
                 echo "Xcode results not found in $BITRISE_XCRESULT_PATH"
                 exit 1
             fi
+            zip -rTy $BITRISE_DEPLOY_DIR/Test.xcresult$(gdate +%Y%m%d%H%M%S).zip $BITRISE_XCRESULT_PATH > /dev/null
```
[the remark this answers is not in the document]
Comment (lpusok) [reply bitrise.yml:243]: It is not related to the changes it only prevents overwriting test artifacts.

File: bitrise.yml (hunk 1 above)
Comment (lpusok) [reply bitrise.yml:53]: This was not intended in itself only is a side result of experimenting with this input

File: main.go (hunk 4 above)
Comment (lpusok) [reply main.go:667]: In case if it is started it will not start again. The headless mode seems not useful other than to show the actual simulator UI and could be removed IMO.

**Review by godrei — APPROVED** [review]
